### DISCUSSION TITLE
[Proposal] "constructor" keyword

### DISCUSSION THREAD

**Discussion by NinjaCross** [dotnet/csharplang#5361] [Language Ideas] [answered]
Personally, I'm all about conciseness, and I think that classes with (very) long names should generally be avoided, 
I have to admit anyway that sometimes a long name is much more usefull than a shorter one, and helps to understand the nature and intention of the object just reading at its name.

I will try to explain why I think we should introduce the keyword "constructor" as an alias for constructors.

Let's say we have this class, with a very long name

```csharp
public class MyClassWithVeryLongNameBecauseItsBetterForComprehension
{
}
```

Unfortunately, long class names have an undesired side-effect: they imply the creation of long-named constructors too.

```csharp
public class MyClassWithVeryLongNameBecauseItsBetterForComprehension
{
  public MyClassWithVeryLongNameBecauseItsBetterForComprehension()
  {
  }
}
```

This is acceptable for constructors without parameters, but it's less than perfect for constructors with (multiple) parameters.

```csharp
public class MyClassWithVeryLongNameBecauseItsBetterForComprehension
{
  public MyClassWithVeryLongNameBecauseItsBetterForComprehension(string parameter1, int parameter2, bool parameter3, char parameter4, float parameter5)
  {
  }
}
```

In this case, in we are forced to word-wrap the parameters into multiple lines in order to avoid horizontal scrolling, leading to a somehow less concise (and monitor unfriendly) representation of the constructor signature

```csharp
public class MyClassWithVeryLongNameBecauseItsBetterForComprehension
{
  public MyClassWithVeryLongNameBecauseItsBetterForComprehension(
      string parameter1, 
      int parameter2, 
      bool parameter3, 
      char parameter4, 
      float parameter5)
  {
  }
}
```

This could be easily avoided using a special, shorter keyword:
- "constructor" (like ECMAScript already does)
- "ctor"
- ".ctor" (like MSIL)

```csharp
public class MyClassWithVeryLongNameBecauseItsBetterForComprehension
{
  public constructor(string parameter1, int parameter2, bool parameter3, char parameter4, float parameter5)
  {
  }
}
```

As I mentioned, this is not someting new, it's already present in other languages, so it should be a familiar concept to a lot of coders.

What do you think ?

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/discussions/1928.

**Reply by NinjaCross** [reply]
You are right, I opened this discussion because I couldn't find a match to my proposal, sorry.

**Reply by Richiban** [reply]
Yeah, GitHub search isn't the best
